.github/workflows/release.yaml

```diff
@@ -23,3 +23,46 @@ jobs:
           args: release --clean
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build-and-push:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Setup Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.ROLE_TO_ASSUME }}
+          role-session-name: "mackerel-sql-metric-collector:${{ github.run_id }}-${{ github.run_number }}"
+          aws-region: ap-northeast-1
+      - name: Login to Public ECR
+        uses: docker/login-action@v3
+        with:
+          registry: public.ecr.aws
+        env:
+          AWS_REGION: us-east-1
+
+      - name: Get version from tag
+        id: get-version
+        run: echo "VERSION=${GITHUB_REF/regs\/tags\//}" >> "$GITHUB_OUTPUT"
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          context: .
+          file: Dockerfile
+          platforms: |
```

Comment:
[may] Multi-arch build on GitHub Actions x86_64 runner could be slow. Using Go's cross-compilation would reduce steps that require QEMU and accelerate the image build. cf.) https://blog.arthur1.dev/entry/2024/01/14/174600

Reply:
Support cross-compilation: https://github.com/mackerelio-labs/mackerel-sql-metric-collector/pull/6/commits/d0d6af5abd1c5df940092f51387399ef93ae9f19